bciers/apps/reporting/src/app/components/changeReview/templates/facilityReportOrganizer.ts

```diff
@@ -0,0 +1,354 @@
+    sourceType.units[parsed.unitIndex] = {
+      unitIndex: parsed.unitIndex,
+      fields: [],
+      fuels: {},
+      emissions: {},
+    };
+  }
+
+  const unit = sourceType.units[parsed.unitIndex];
+
+  if (parsed.fuelIndex !== undefined) {
+    if (!unit.fuels[parsed.fuelIndex]) {
+      unit.fuels[parsed.fuelIndex] = {
+        fuelIndex: parsed.fuelIndex,
+        fields: [],
+        emissions: {},
+      };
+    }
+
+    if (parsed.emissionIndex !== undefined) {
+      if (!unit.fuels[parsed.fuelIndex].emissions[parsed.emissionIndex]) {
+        unit.fuels[parsed.fuelIndex].emissions[parsed.emissionIndex] = {
+          emissionIndex: parsed.emissionIndex,
+          fields: [],
+        };
+      }
+      unit.fuels[parsed.fuelIndex].emissions[parsed.emissionIndex].fields.push(
+        change,
+      );
+    } else {
+      unit.fuels[parsed.fuelIndex].fields.push(change);
+    }
+  } else if (parsed.emissionIndex !== undefined) {
+    if (!unit.emissions[parsed.emissionIndex]) {
+      unit.emissions[parsed.emissionIndex] = {
+        emissionIndex: parsed.emissionIndex,
+        fields: [],
+      };
+    }
+    unit.emissions[parsed.emissionIndex].fields.push(change);
+  } else {
+    unit.fields.push(change);
+  }
+}
+
+function parseActivityData(
+  facility: FacilityReportStructure,
+  parsed: any,
+  change: ChangeItem,
+) {
+  // Type guard to ensure we have the right structure
+  if (typeof change.new_value !== "object" || !change.new_value?.source_types)
+    return;
+
+  Object.entries(change.new_value.source_types).forEach(
+    ([sourceTypeName, sourceTypeData]: [string, any]) => {
+      const activity = facility.activities[parsed.activityName];
+
+      if (!activity.sourceTypes[sourceTypeName]) {
+        activity.sourceTypes[sourceTypeName] = {
+          sourceTypeName,
+          fields: [],
+          units: {},
+        };
+      }
+
+      const sourceType = activity.sourceTypes[sourceTypeName];
+
+      if (sourceTypeData.units && Array.isArray(sourceTypeData.units)) {
+        sourceTypeData.units.forEach((unitData: any, unitIndex: number) => {
+          if (!sourceType.units[unitIndex]) {
+            sourceType.units[unitIndex] = {
+              unitIndex,
+              fields: [],
+              fuels: {},
+              emissions: {},
+            };
+          }
+
+          const unit = sourceType.units[unitIndex];
+
+          // Add unit-level fields
+          Object.entries(unitData).forEach(([key, value]) => {
+            if (key !== "fuels" && key !== "emissions") {
+              unit.fields.push({
+                deletedActivities: undefined,
+                facilityName: undefined,
+                field: `${change.field}[source_types][${sourceTypeName}][units][${unitIndex}][${key}]`,
+                old_value: change.old_value as string | Record<string, any>,
+                new_value: value as string | Record<string, any>,
+                change_type: "added",
```

Comment:
same question as above - how do we know that it's been added (especially since we may have a value for `change.old_value`?